lgsm/functions/command_monitor.sh

```diff
@@ -11,76 +11,79 @@ commandaction="Monitoring"
 functionselfname="$(basename "$(readlink -f "${BASH_SOURCE[0]}")")"
 fn_firstcommand_set
 
+fn__restart_server() {
+	if [ -f "${lockdir}/${selfname}.lock" ]; then
+		alert="${1:?}}"
+		alert.sh
+		(
+			fn_print_info_nl "Restarting the server and skip immediate exit"
+			exitbypass=1
+			command_stop.sh
+			command_start.sh
+			echo "" # start doesn't always print newline
+		)
+		fn_firstcommand_reset
+	else
+		fn_print_warn_nl "Skipping server restart because lockfile is missing, probably server is restarted during monitor execution."
+	fi
+}
+
 fn_monitor_check_lockfile(){
-	# Monitor does not run it lockfile is not found.
+	fn_print_dots "Checking lockfile"
+
+	# Monitor does not run if lockfile is not found.
 	if [ ! -f "${lockdir}/${selfname}.lock" ]; then
-		fn_print_dots "Checking lockfile: "
-		fn_print_checking_eol
-		fn_script_log_info "Checking lockfile: CHECKING"
-		fn_print_error "Checking lockfile: No lockfile found: "
-		fn_print_error_eol_nl
-		fn_script_log_error "Checking lockfile: No lockfile found: ERROR"
-		echo -e "* Start ${selfname} to run monitor."
+		fn_print_fail_nl "Checking lockfile: No lockfile found"
+		exitcode="3"
 		core_exit.sh
-	fi
-
+	
 	# Fix if lockfile is not unix time or contains letters
-	if [ -f "${lockdir}/${selfname}.lock" ]&&[[ "$(head -n 1 "${lockdir}/${selfname}.lock")" =~ [A-Za-z] ]]; then
+	elif [[ "$(head -n 1 "${lockdir}/${selfname}.lock")" =~ [A-Za-z] ]]; then
+		fn_print_warn_nl "Checking lockfile: fixing illegal lockfile"
 		date '+%s' > "${lockdir}/${selfname}.lock"
 		echo "${version}" >> "${lockdir}/${selfname}.lock"
 		echo "${port}" >> "${lockdir}/${selfname}.lock"
+	else
+		fn_print_ok_nl "Checking lockfile"
 	fi
 }
 
 fn_monitor_check_update(){
+	fn_print_dots "Checking active updates"
+
 	# Monitor will check if update is already running.
 	if [ "$(pgrep "${selfname} update" | wc -l)" != "0" ]; then
-		fn_print_dots "Checking active updates: "
-		fn_print_checking_eol
-		fn_script_log_info "Checking active updates: CHECKING"
-		fn_print_error_nl "Checking active updates: SteamCMD is currently checking for updates: "
-		fn_print_error_eol
-		fn_script_log_error "Checking active updates: SteamCMD is currently checking for updates: ERROR"
```

Comment:
What is your rule for using `fn_script_log_*` here ?
Do you prefer setting the exitcode by fn_scrip_log_* in this case I can add it also in the other locations where I set exitcode directly.

I didn't add it because it seems like duplicate log output, if there would be a fn_script_log_dots [and replacing in monitor fn_print_dots with it] or fn_print_*_nl() which also setting the exitcode, I would be more satisfied. Would you be fine with such a solution?